graphix/visualization.py

```diff
@@ -1014,8 +1014,8 @@ def get_pos_from_flow(self, f: dict[int, int], l_k: dict[int, int]) -> dict[int,
         for i, k in enumerate(start_nodes):
             pos[k][1] = i
             while k in f.keys():
-                k = list(f[k])[0]
-                pos[k][1] = i
+                kk = next(iter(f[k]))
+                pos[kk][1] = i
```

Comment:
The purpose of this `while` loop is to make `k` follow the path given by `f`: it is essential for `k` to be updated during the loop, or else the loop never terminates (because `k` will never change). If the purpose of this change is to prevent updating a `for`-loop variable, a solution could be:
```python
            node = k
            while node in f.keys():
                node = next(iter(f[node]))
                pos[node][1] = i
```

Reply:
(Note: we are missing `test_visualization.py` to catch that!)

Reply:
> the purpose of this change is to prevent updating a for-loop variable

This is exactly what I meant.
Anyway we need to cover these lines...

Reply:
I propose a fix and a test here: https://github.com/thierry-martinez/graphix/commit/d761465b71995674193920dcbd0332fb0b4ee57d